src/test/unistd/test_unistd.c

```diff
@@ -1,118 +0,0 @@
-#include <glib.h>
-#include <signal.h>
-#include <stdbool.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <sys/utsname.h>
-#include <unistd.h>
-
-#include "test/test_common.h"
-#include "test/test_glib_helpers.h"
-
-// Tests that the results are plausible, but can't really validate that it's our
-// pid without depending on other functionality.
-static void _test_getpid_nodeps() {
-    int pid = getpid();
-    g_assert_cmpint(pid,>,0);
-    g_assert_cmpint(getpid(),==,pid);
-}
-
-// Must be declared volatile because it is modified in the signal handler.
-static volatile int sigaction_count = 0;
-static void sigaction_inc(int sig) {
-    ++sigaction_count;
-}
-
-// Validates that the returned pid is ours by using it to send a signal to
-// ourselves.
-static void _test_getpid_kill() {
-    struct sigaction action = {.sa_handler = sigaction_inc};
-    assert_nonneg_errno(sigaction(SIGUSR1, &action, NULL));
-
-    int pid = getpid();
-
-    sigaction_count = 0;
-    assert_nonneg_errno(kill(pid, SIGUSR1));
-    g_assert_cmpint(sigaction_count, ==, 1);
-}
-
-static void _test_gethostname(gconstpointer gp_nodename) {
-    const char* nodename = gp_nodename;
-    char buf[1000] = {0};
-
-    // Invalid pointer. Documented to return errno=EFAULT in gethostname(2),
-    // but segfaults on Ubuntu 18.
-    //
-    // g_assert_cmpint(gethostname(NULL+1, 100),==,-1);
-    // assert_errno_is(EFAULT);
-
-    // Negative len. Documented to return errno=EINVAL in gethostname(2), but
-    // segfaults on Ubuntu 18.
-    //
-    // g_assert_cmpint(gethostname(buf, -1),==,-1);
-    // assert_errno_is(EINVAL);
```

Comment:
Since these tests don't work, you don't need to convert them to Rust, but I think it would still be useful to move these comments to the Rust version so we know why we aren't testing them.